### PR TITLE
Rhospec uses genesis

### PR DESCRIPTION
**Pull request by ArturGajowy**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

Make all RhoSpec run with all genesis contracts

This will also make less likely the mistake of registering a contract
in StandardDeploys but not putting it in Genesis - which I've done twice
already.

This is also a first step for customizing the genesis block - including
initial ~~wallets~~ vaults - for purposes of particular tests.

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/test/scala/coop/rchain/casper/genesis/contracts/TestUtil.scala [hunk 1]
```diff
@@ -35,57 +29,68 @@ object TestUtil {
...
+      runtime <- TestRuntime.create[F, G](extraServices)
+      _       <- Runtime.injectEmptyRegistryRoot[F](runtime.space, runtime.replaySpace)
+    } yield runtime
+  }
+
+  def runTestsWithDeploys[F[_]: Concurrent: ContextShift, G[_]: Parallel[F, ?[_]]](
+      tests: CompiledRholangSource,
+      genesisSetup: RuntimeManager[F] => F[BlockMessage],
+      otherLibs: Seq[DeployData],
+      additionalDSystemProcesses: Seq[SystemProcess.Definition[F]]
```
Comment (lukasz-golebiewski) [inline casper/src/test/scala/coop/rchain/casper/genesis/contracts/TestUtil.scala]: `additionalD` looks like a typo

File: casper/src/test/scala/coop/rchain/casper/genesis/contracts/TimeoutResultCollectorSpec.scala [hunk 2]
```diff
@@ -10,7 +10,7 @@ class TimeoutResultCollectorSpec extends FlatSpec with AppendedClues with Matche
   it should "testFinished should be false if execution hasn't finished within timeout" in {
```
Comment (lukasz-golebiewski) [inline casper/src/test/scala/coop/rchain/casper/genesis/contracts/TimeoutResultCollectorSpec.scala]: This reads awkwardly. Can you change to `"testFinished" should "be false...`?

File: casper/src/test/scala/coop/rchain/casper/genesis/contracts/TimeoutResultCollectorSpec.scala [hunk 3]
```diff
@@ -10,7 +10,7 @@ class TimeoutResultCollectorSpec extends FlatSpec with AppendedClues with Matche
...
     RhoSpec
       .getResults(CompiledRholangSource("TimeoutResultCollectorTest.rho"), Seq.empty)
-      .runSyncUnsafe(Duration.Zero)
+      .runSyncUnsafe(5.seconds)
```
Comment (lukasz-golebiewski) [inline casper/src/test/scala/coop/rchain/casper/genesis/contracts/TimeoutResultCollectorSpec.scala]: Why is this change needed?

File: casper/src/test/scala/coop/rchain/casper/util/rholang/Interactive.scala [hunk 4]
```diff
@@ -53,7 +55,7 @@ class Interactive private (runtime: Runtime[Task])(implicit scheduler: Scheduler
   def tuplespace: String = StoragePrinter.prettyPrint(runtime.space.store)
 
   def eval(code: String): Unit = {
-    TestUtil.eval(code, runtime)
+    TestUtil.eval(code, runtime).runSyncUnsafe(Duration.Inf)
```
Comment (lukasz-golebiewski) [inline casper/src/test/scala/coop/rchain/casper/util/rholang/Interactive.scala]: This may cause tests to hang in case of deadlocks. Better use a finite duration

File: casper/src/main/scala/coop/rchain/casper/genesis/contracts/Rev.scala [hunk 5]
```diff
@@ -18,6 +18,8 @@ class Rev[A](
   private val minimumBond = posParams.minimumBond
   private val maximumBond = posParams.maximumBond
 
+  final val path = "<synthetic in Rev.scala>"
```
Comment (lukasz-golebiewski) [inline casper/src/main/scala/coop/rchain/casper/genesis/contracts/Rev.scala]: Why isn't this optional?

File: casper/src/test/scala/coop/rchain/casper/genesis/contracts/TimeoutResultCollectorSpec.scala (hunk 2 above)
Comment (ArturGajowy) [reply casper/src/test/scala/coop/rchain/casper/genesis/contracts/TimeoutResultCollectorSpec.scala]: Not related with this PR

File: casper/src/test/scala/coop/rchain/casper/genesis/contracts/TimeoutResultCollectorSpec.scala (hunk 3 above)
Comment (ArturGajowy) [reply casper/src/test/scala/coop/rchain/casper/genesis/contracts/TimeoutResultCollectorSpec.scala]: See commit message

File: casper/src/test/scala/coop/rchain/casper/util/rholang/Interactive.scala (hunk 4 above)
Comment (ArturGajowy) [reply casper/src/test/scala/coop/rchain/casper/util/rholang/Interactive.scala]: This isn't used for tests, but for interactive usage

File: casper/src/main/scala/coop/rchain/casper/genesis/contracts/Rev.scala (hunk 5 above)
Comment (ArturGajowy) [reply casper/src/main/scala/coop/rchain/casper/genesis/contracts/Rev.scala]: Because you want to know even if it's synthetic (generated) source

Comment (ArturGajowy) [reply casper/src/main/scala/coop/rchain/casper/genesis/contracts/Rev.scala]: I do, anyway

Comment (lukasz-golebiewski) [reply casper/src/main/scala/coop/rchain/casper/genesis/contracts/Rev.scala]: Know what exactly? 
The characters currently residing under the `path` val express the fact that for this particular class a `path` doesn't exist. This can be expressed isomorphically with a `None`

File: casper/src/test/scala/coop/rchain/casper/genesis/contracts/TimeoutResultCollectorSpec.scala (hunk 2 above)
Comment (lukasz-golebiewski) [reply casper/src/test/scala/coop/rchain/casper/genesis/contracts/TimeoutResultCollectorSpec.scala]: Loosely related, hence the question. No need to follow the boy scout rule every time

File: casper/src/test/scala/coop/rchain/casper/util/rholang/Interactive.scala (hunk 4 above)
Comment (lukasz-golebiewski) [reply casper/src/test/scala/coop/rchain/casper/util/rholang/Interactive.scala]: I see! In that case instead of introducing yet another `unsafe` operation I would opt for introducing a for comprehension here

Comment (ArturGajowy) [reply casper/src/test/scala/coop/rchain/casper/util/rholang/Interactive.scala]: The unsafe was always there, under the `Unit` returned previously. I'd fix it if I was going to work with that code soon, but I don't think so :)

File: casper/src/main/scala/coop/rchain/casper/genesis/contracts/Rev.scala (hunk 5 above)
Comment (ArturGajowy) [reply casper/src/main/scala/coop/rchain/casper/genesis/contracts/Rev.scala]: the path, what else? this field is purely for debuggability purposes. It's not iso with None, as once s/b generates more .rho in scala, they need to provide the path.
